### PR TITLE
LegalPath: change defines to an enum class

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Change a family of defines into an enum class

## Motivation and context
Less preprocessor, more type safety

## How has this been tested?
It builds..

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/DllLoader/exports/emu_msvcrt.cpp [hunk 1]
```diff
@@ -990,7 +990,8 @@ extern "C"
         strncpy(entry->d_name, "..\0", 3);
       else
       {
-        strncpy(entry->d_name, dirData->items[dirData->curr_index - 2]->GetLabel().c_str(), sizeof(entry->d_name));
+        strncpy(entry->d_name, dirData->items[dirData->curr_index - 2]->GetLabel().c_str(),
+                sizeof(entry->d_name) - 1);
```
Comment (fuzzard) [inline xbmc/cores/DllLoader/exports/emu_msvcrt.cpp:994]: intended to slip this in as well?

**Review by fuzzard — APPROVED** [review]
Couple oddities that dont seem related. but minors, so whatever.

File: xbmc/cores/DllLoader/exports/emu_msvcrt.cpp (hunk 1 above)
Comment (notspiff) [reply xbmc/cores/DllLoader/exports/emu_msvcrt.cpp]: nope, removed

File: xbmc/utils/LegacyPathTranslation.cpp [hunk 2]
```diff
@@ -89,7 +89,9 @@ std::string CLegacyPathTranslation::TranslateMusicDbPath(const std::string &lega
   return TranslatePath(legacyPath, s_musicDbTranslator, MusicDbTranslatorSize);
 }
 
-std::string CLegacyPathTranslation::TranslatePath(const std::string &legacyPath, Translator *translationMap, size_t translationMapSize)
+std::string CLegacyPathTranslation::TranslatePath(const std::string& legacyPath,
+                                                  Translator* translationMap,
+                                                  size_t translationMapSize)
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/utils/LegacyPathTranslation.cpp]: ide doing stupid things and then me doing stupid things. removed.

**Pull request merged**